wavefront/server/plugins/datasource/datasource/postgres/__init__.py

```
@@ -0,0 +1,164 @@
+
+            query_to_execute = query_to_execute.replace(
+                '{{rls}}', f'{odata_data_filter}' if odata_data_filter else 'TRUE'
+            )
+            query_to_execute = query_to_execute.replace(
+                '{{filters}}', f'{odata_filter}' if odata_filter else 'TRUE'
+            )
+            query_to_execute = query_to_execute.rstrip().rstrip(';')
+            depth = 0
+            has_top_level_limit = False
+            for token in re.split(r'(\(|\))', query_to_execute):
+                if token == '(':
```

Ruff (0.15.10)

[error] Possible hardcoded password assigned to: "token"

(S105)